### ISSUE TITLE
Setup fails on Windows 10 

### ISSUE THREAD

**Issue by markus237** [msys2/MSYS2-packages#2364]
Hello,
I have the following issue after installing msys2-x86_64-20210228 for the first time. 

> mkdir: cannot create directory ‘’: No such file or directory
> gpg: marginals needed: 3  completes needed: 1  trust model: pgp
> gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
> chmod: cannot access '/trustdb.gpg': No such file or directory
> /usr/bin/pacman-key: line 216: ((: >= 0 : syntax error: operand expected (error token is ">= 0 ")
> ==> ERROR: You do not have sufficient permissions to read the pacman keyring.
> ==> Use 'pacman-key --init' to correct the keyring permissions.
> ==> ERROR: You do not have sufficient permissions to read the pacman keyring.
> ==> Use 'pacman-key --init' to correct the keyring permissions.

These errors occur directly after installation and everytime I open msys2 again.
I tried multiple msys Version > 2018 and the errors are the same.
On Windows 7 it works.

OS:
Windows 10 Enterprise version 1909 64bit

Thanks for any help!

**Comment by pischky** [comment]
Is this a problem with permission? 
Does it also happen when you run msys2-x86_64-20210228.exe as Administrator?
We have had similarly problems running as normal user.

**Comment by Biswa96** [comment]
msys2 does not require administrator permission. Also I'd suggest not to run it as admin.

**Comment by pischky** [comment]
Yes. But we have a problem that on Win10 after installation with msys2-x86_64-20210228.exe (as Administrator) we where not able to execute pacman via shell started from normal user on a Win10 System. On Win7 this worked.

**Comment by Novantric** [comment]
Yeah but that doesn't help.
I've reinstalled countless times and it's always this same error, preventing me from doing anything.
I just run the installation as normal.

**Comment by Biswa96** [comment]
Can you share the error output from the terminal window?

**Comment by ftab** [comment]
I have the same issue. Windows 10, installed msys2-x86_64-20230718.exe, the terminal gave this error immediately:

```
mkdir: cannot create directory ‘’: No such file or directory
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
chmod: cannot access '/trustdb.gpg': No such file or directory
/usr/bin/pacman-key: line 244: ((: >= 0 : syntax error: operand expected (error token is ">= 0 ")
==> ERROR: You do not have sufficient permissions to read the pacman keyring.
==> Use 'pacman-key --init' to correct the keyring permissions.
==> ERROR: You do not have sufficient permissions to read the pacman keyring.
==> Use 'pacman-key --init' to correct the keyring permissions.
```

I tried to follow its instruction but got basically the same error:
```
# pacman-key --init
mkdir: cannot create directory ‘’: No such file or directory
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
chmod: cannot access '/trustdb.gpg': No such file or directory
/usr/bin/pacman-key: line 244: ((: >= 0 : syntax error: operand expected (error token is ">= 0 ")
```

The file it's talking about, trustdb.gpg, exists under C:\msys64\home\fury\.gnupg\ but I'm not sure why it's trying to access it at the root.

**Comment by benoit-germain** [comment]
I have the same error with mys2-x86-64-20240727.exe.
It seems like PACMAN_KEYRING_DIR is empty, but I don't know where pacman-key is supposed to obtain it if not forced with option --gpgdir.
This option is not enough to get rid of all errors though:
```
# pacman-key --gpgdir ~/.gnupg --init
/usr/bin/pacman-key: line 245: ((: >= 0 : syntax error: operand expected (error token is ">= 0 ")
gpg: starting migration from earlier GnuPG versions
gpg: porting secret keys from '/c/Users/bgermain/.gnupg/secring.gpg' to gpg-agent
gpg: migration succeeded
```

It looks like vercmp misbehaves somehow. According to man, I should see something like
```
$ vercmp 1 2
-1
```

but what I have instead is
```
# vercmp --help

# vercmp 1 2

```
